scipy/signal/spectral.py

```diff
@@ -2003,3 +2004,181 @@ def _median_bias(n):
+
+    .. math:: H = (1+\\alpha)/2
+
+    .. versionadded:: 1.4.0
+
+    See Also
+    ---------
+    signal.waveforms.coloured_noise : Generate data with defined Hurst exponent
+    np.linalg.lstsq : Description of the "full" parameter
+
+    References
+    ----------
+    .. [1] Peng, C‐K., et al. "Quantification of scaling exponents and crossover
+           phenomena in nonstationary heartbeat time series." Chaos: An
+           Interdisciplinary Journal of Nonlinear Science 5.1 (1995): 82-87.
+
+    .. [2] Kantelhardt, Jan W., et al. "Detecting long-range correlations with
+           detrended fluctuation analysis." Physica A: Statistical Mechanics
+           and its Applications 295.3-4 (2001): 441-454.
+    """
+
+    if s_min <= 0:
+        raise ValueError("s_min must be positive")
+    if s_max >= len(x):
+        raise ValueError("s_max cannot be longer than x. Decrease s_max")
+    if s_max <= s_min:
+        raise ValueError("s_max must be smaller than s_min")
+    if s_min == 1:
+        warnings.warn('Fitting is inaccurate for s_min == 1. Increase s_min')
+
+    # Pre-allocate array of f_k
+    f_k_s = np.zeros((len(x), s_max))
+    mean_x = np.mean(x)
+
+    for s in range(s_min, s_max + 1):
+        # Pre-allocate/reset inner loops
+        fitrange = np.arange(start=1, stop=len(x) - s + 1, step=s)
+        delta_y_s = np.zeros(fitrange[-1] + s - 1)
+        f_k_substep = np.zeros(np.floor(len(delta_y_s) / s).astype(int))
+        s_axis = np.arange(1, s + 1)
+
+        # Sum diff for i=1 to N and j=1 to i and ensure integer num of windows
+        delta_y = np.cumsum(x - mean_x)[:len(delta_y_s)]
+
+        # Perform linear fits
+        for fitx in range(0, len(delta_y), s):
+            polyco = np.polyfit(s_axis, delta_y[fitx:fitx + s], 1)
+            delta_y_s[fitx:fitx + s] = (s_axis * polyco[0]) + polyco[1]
+
+        # Calculate RMS fluctuations of linear detrending
+        for k_range in range(len(f_k_substep)):
+            sumstart = (k_range * s)
+            sumend = (k_range * s) + s
+
+            f_k_substep[k_range] = sum((delta_y[sumstart:sumend] -
+                                        delta_y_s[sumstart:sumend]) ** 2)
+        f_k_s[:len(f_k_substep), s - 1] = np.sqrt((1 / s) * f_k_substep)
+
+    # Calculate mean (ignoring leftover values = 0).
+    f_k_s[f_k_s == 0] = np.nan
+    f_s = np.nanmean(f_k_s[:, s_min - 1:], axis=0)
+
+    # Calculate Hurst exponent
+    log_f_s = np.log(f_s)
+    log_s_axis = np.log(np.arange(s_min, s_max + 1))
+    c, resids, rank, s, rcond = np.polyfit(log_s_axis, log_f_s, 1, full=True)
+    hurst = c[0]
+
+    # Plot
+    if plot:
+        import matplotlib.pyplot as plt
+        plt.plot(log_s_axis, log_f_s, 'x')
+        plt.plot(log_s_axis, (c[0] * log_s_axis) + c[1], 'k')
+        plt.title('Hurst Exponent')
+        plt.xlabel(r'Log (Window Size, $s$)')
+        plt.xlim(0)
+        plt.ylabel(r'Log ($F_s$)')
+        plt.show()
+
+    if full:
```

Comment:
This type of variable number of return arguments are problematic. Is there a way we can return always the same type of arguments?

Reply:
My intention was to follow the syntax of [np.polyfit](https://docs.scipy.org/doc/numpy/reference/generated/numpy.polyfit.html). 

It may make more sense to always return the covariance matrix, V, of the fit as a second parameter, then do away with the "full" argument entirely. Would this be a better option?

Reply:
(misclick, sorry). Indeed, better avoid these sorts of variable return values. Either return a tuple, or use a Bunch pattern